intrastat_base/models/res_partner.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2022 Noviat.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+XI_COUNTY_NAMES = [
+    "antrim",
+    "armagh",
+    "down",
+    "fermanagh",
+    "londonderry",
+    "tyrone",
+    "northern ireland",
+]
+
+XI_COUNTIES = [
+    "base.state_uk18",  # County Antrim
+    "base.state_uk19",  # County Armagh
+    "base.state_uk20",  # County Down
+    "base.state_uk22",  # County Fermanagh
+    "base.state_uk23",  # County Londonderry
+    "base.state_uk24",  # County Tyrone
+    "base.state_ie_27",  # Antrim
+    "base.state_ie_28",  # Armagh
+    "base.state_ie_29",  # Down
+    "base.state_ie_30",  # Fermanagh
+    "base.state_ie_31",  # Londonderry
+    "base.state_ie_32",  # Tyrone
+]
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def _get_xi_counties(self):
+        return [self.env.ref(x) for x in XI_COUNTIES]
+
+    @api.model
+    def _get_xu_counties(self):
+        uk_counties = self.env.ref("base.uk").state_ids
+        xu_counties = uk_counties.filtered(lambda r: r not in self._get_xi_counties())
+        return xu_counties
+
+    def _get_intrastat_country_code(self, country=None, state=None):
```

Comment:
This method is really strange with it's "if self / else" ! When self has a value, then the arguments country and state are ignored, which is surprising. For me, instead of this wired code, we should have 2 separate methods, the first one calling the second one.